src/compile/data/transforms.ts

```diff
@@ -130,11 +132,24 @@ export function parseTransformArray(model: Model) {
   let previous: DataFlowNode;
   let lookupCounter = 0;
 
+  function insert(newNode: DataFlowNode) {
+    if (!first) {
+      // A parent may be inserted during node construction
+      // (e.g., selection FilterNodes may add a TimeUnitNode).
+      first = newNode.parent || newNode;
```

Comment:
I feel we are already breaking encapsulation by passing a dataflow node into the predicate method. But I guess it makes sense so we can use the same codepath.